index.js

```diff
@@ -191,6 +192,12 @@
         readFromRoute = !!val;
         return this;
       };
+      
+      // Custom route tracking function
+      this.trackRouteFunction = function(trackingFunction) {
+        trackRouteFn = trackingFunction;
```

Comment:
Check that `trackingFunction` is actually a function before assigning it. No need to throw an error if it is not, just don't set trackeRouteFn.